### PR TITLE
Prepare 8.0.0 DR release

### PR DESCRIPTION
**Pull request by johnfxgalea**
Makes the necessary changes to prepare for 8.0.0 release.

### CODE REVIEW

File: api/docs/release.dox [hunk 1]
```diff
@@ -158,7 +174,7 @@ changes:
  - Changed the 32-bit x86 stack alignment of DynamoRIO and clients built using
    DR's CMake configuration from 4 to 16 on Linux to match modern conventions.
 
-The changes between version \DR_VERSION and 7.1.0 include the following minor
+The changes between version 8.0.0 and 7.1.0 include the following minor
 compatibility changes:
 
  - Added the define #MCXT_NUM_SIMD_SLOTS that was renamed from NUM_SIMD_SLOTS.
```
Comment (derekbruening) [inline api/docs/release.dox:180]: (It won't let me put a comment any further down)
If you could add to the list of non-compat changes sthg like this:

 - Fixed major synchronization bugs on ARM and AArch64.

And maybe include that in the description of the final Github Release.

File: api/docs/release.dox [hunk 2]
```diff
@@ -133,7 +133,23 @@ Dr. Memory Framework (DRMF) in the same package as DynamoRIO.  DRMF
 provides the umbra, drsyscall, and drsymcache Extensions for use by
 clients.
 
-The changes between version \DR_VERSION and 7.1.0 include the following compatibility
+The changes between version \DR_VERSION and 8.0.0 include the following compatibility
+changes:
+
```
Comment (derekbruening) [inline api/docs/release.dox:138]: suggest: I usually put one line `- Nothing yet.` in these sections so it's very clear how to edit.

File: api/docs/release.dox (hunk 1 above)
Comment (johnfxgalea) [reply api/docs/release.dox:180]: Done. What description are you referring to please?

Comment (derekbruening) [reply api/docs/release.dox:186]: This one: https://github.com/DynamoRIO/dynamorio/wiki/New-Release#update-the-release-description
I.e., the one that shows up on a releases/ page (this one just says "Auto-generated periodic build."): https://github.com/DynamoRIO/dynamorio/releases/tag/cronbuild-7.91.18369

Comment (derekbruening) [reply api/docs/release.dox:186]: I just updated that wiki section: normally I've been pointing at the changelog, as wsa done here: https://github.com/DynamoRIO/dynamorio/releases/tag/release_7.1.0.  I would say, do that, and also list the key highlights inlined on the releases page.  Maybe sthg like:

- Added Win10 1809 and higher support.
- Added AVX-512 support.
- Fixed major synchronization bugs on ARM and AArch64.
- Changed 32-bit Linux x86 stack alignment to 16.

Plus anything else you want to call out: those are the major ones that come to mind.

Comment (johnfxgalea) [reply api/docs/release.dox:186]: drbbdup? :)

Comment (derekbruening) [reply api/docs/release.dox:186]: Yes -- I don't see it in release.dox though: I would expect it to be there too.
Maybe also drdisas, and drcachesim function tracing enhancements.

Scanning the list I also see a typo:
```
 - Added -record_replace_retaddrd_only to drcachesim for faster function tracing
```
s/retaddrd/retaddr/

There are many small API additions that we can leave to the full changelist.

Comment (johnfxgalea) [reply api/docs/release.dox:186]: It's there: " Added a new DR extension, namely "drbbdup", which enables different case
   instrumentation of the same basic block by duplicating code."

Comment (derekbruening) [reply api/docs/release.dox:186]: Oh it is in the wrong place -- it's down under the 7.1.0 new features!  That is why I did not see it.

Comment (johnfxgalea) [reply api/docs/release.dox:186]: Fixed!

**Pull request merged**